`magma` build failure on x86_64-linux as of `4c344da2`

## Issue description
Build of `magma` failed on x86_64-linux as of 4c344da29a5b46caadb87df1d194082a190e1199. This is currently breaking `python3Packages.pytorchWithCuda`.

```
error: builder for '/nix/store/0z162w2f5vdzbmxk9nwynygcjzbrhh7h-magma-2.5.4.drv' failed with exit code 1;
       last 10 log lines:
       >
       > /nix/store/pvn23vycg674bj6nypjcfyhqbr85rqxa-glibc-2.34-115-dev/include/stdlib.h(797): error: attribute "__malloc__" does not take arguments
       >
       > 13 errors detected in the compilation of "/build/magma-2.5.4/magmablas/zaxpycp.cu".
       > CMake Error at magma_generated_zaxpycp.cu.o.Release.cmake:280 (message):
       >   Error generating file
       >   /build/magma-2.5.4/build/CMakeFiles/magma.dir/magmablas/./magma_generated_zaxpycp.cu.o
       >
       >
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/0z162w2f5vdzbmxk9nwynygcjzbrhh7h-magma-2.5.4.drv'.
error: 1 dependencies of derivation '/nix/store/8z8wgh4n3w7pwrmr3mbwh63m2iijpcpa-python3.9-pytorch-1.11.0.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2199000648)

python3Packages.pytorchWithCuda maintainers cc: @teh @thoughtpolice @tscholak
magma maintainers cc: @tbenst
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 4c344da29a5b46caadb87df1d194082a190e1199
2. Run `nix-build -A magma`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1021-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - nixpkgs: `/nix/store/r902mrj2iwsjhy8gs6y0rbr9nada0z4z-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2199000648) that created this issue.
- Internal tag: 5878480090433585821

--- a/pkgs/development/libraries/science/math/magma/default.nix
+++ b/pkgs/development/libraries/science/math/magma/default.nix
@@ -8,7 +8,7 @@ assert let majorIs = lib.versions.major cudatoolkit.version;
        in majorIs == "9" || majorIs == "10" || majorIs == "11";
 
 let
-  version = "2.5.4";
+  version = "2.6.2";
 
   # We define a specific set of CUDA compute capabilities here,
   # because CUDA 11 does not support compute capability 3.0. Also,
@@ -45,7 +45,7 @@ in stdenv.mkDerivation {
   inherit version;
   src = fetchurl {
     url = "https://icl.cs.utk.edu/projectsfiles/magma/downloads/magma-${version}.tar.gz";
-    sha256 = "0rrvd21hczxlm8awc9z54fj7iqpjmsb518fy32s6ghz0g90znd3p";
+    hash = "sha256-dbVU2rAJA+LRC5cskT5Q5/iMvGLzrkMrWghsfk7aCnE=";
     name = "magma-${version}.tar.gz";
   };
 
@@ -63,23 +63,6 @@ in stdenv.mkDerivation {
 
   enableParallelBuilding=true;
   buildFlags = [ "magma" "magma_sparse" ];
-
-  # MAGMA's default CMake setup does not care about installation. So we copy files directly.
-  installPhase = ''
-    mkdir -p $out
-    mkdir -p $out/include
-    mkdir -p $out/lib
-    mkdir -p $out/lib/pkgconfig
-    cp -a ../include/*.h $out/include
-    #cp -a sparse-iter/include/*.h $out/include
-    cp -a lib/*.so $out/lib
-    cat ../lib/pkgconfig/magma.pc.in                   | \
-    sed -e s:@INSTALL_PREFIX@:"$out":          | \
-    sed -e s:@CFLAGS@:"-I$out/include":    | \
-    sed -e s:@LIBS@:"-L$out/lib -lmagma -lmagma_sparse": | \
-    sed -e s:@MAGMA_REQUIRED@::                       \
-        > $out/lib/pkgconfig/magma.pc
-  '';
 
   meta = with lib; {
     description = "Matrix Algebra on GPU and Multicore Architectures";

